src/vec.rs

```diff
@@ -606,6 +629,27 @@ where
             .expect("Replaced old value");
     }
 
+    /// Try to add the given element to the `identified_vec` unconditionally, either appending it to the `identified_vec``, or
+    /// replacing an existing value if it's already present.
+    ///
+    /// - Parameter item: The value to append or replace.
+    /// - Returns: A Result with either the original element that was replaced by this operation, or `None` if the value was
+    ///   appended to the end of the collection. If the given element already exists within the collection the function call returns a InsertionFailure.
+    /// - Complexity: The operation is expected to perform amortized O(1) copy, hash, and compare
+    ///   operations on the `ID` type, if it implements high-quality hashing.
+    #[inline]
+    pub fn try_update_or_append(
```

Comment:
hmm I thought we wanted a corresponding method like this: https://github.com/radixdlt/babylon-wallet-ios/blob/c69d2ca45ca5435f13a271496b7e585368dc4713/RadixWallet/Profile/Entity/PersonaData/PersonaData%2BCollectionOfIdentifiedEntries.swift#L29

which should Fail **if the element does NOT exist**. so a `try_update` which will throw `Error::ExpectedElementNotPresent` error?  

Because this method makes no sense... it will fail if it does already exist, then it calles `update` (which it cannot do...)

Reply:
You are entirely correct. The function didn't make sense but it has now been rectified.